src/main/java/dev/sigstore/rekor/client/HashedRekordRequest.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2022 The Sigstore Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package dev.sigstore.rekor.client;
+
+import dev.sigstore.json.GsonSupplier;
+import dev.sigstore.rekor.*;
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+import java.security.cert.Certificate;
+import java.util.Base64;
+import java.util.HashMap;
+import org.bouncycastle.openssl.jcajce.JcaPEMWriter;
+import org.bouncycastle.util.encoders.Hex;
+
+public class HashedRekordRequest {
+
+  private final Hashedrekord hashedrekord;
+
+  private HashedRekordRequest(Hashedrekord hashedrekord) {
+    this.hashedrekord = hashedrekord;
+  }
+
+  /**
+   * Create a new HashedRekorRequest.
+   *
+   * @param artifactDigest the sha256 digest of the artifact (not hex/base64 encoded)
+   * @param leafCert the leaf certificate used to verify {@code signature}, usually obtained from
```

Comment:
for custom key signed artifacts (not using fulcio), is there a leaf cert to submit, or do you just use the singer's public key? It's all still a little muddled for me so asking out of curiosity,